decam/makeLinearizer.py

```diff
@@ -61,6 +62,33 @@ def makeLinearizerDecam(fromFile, force=False, verbose=False):
                 print("LSST  table for %s=%s..." % (ampName, lsstTable[i, 0:5],))
         linearizer = lsstTable
         butler.put(linearizer, "linearizer", dataId=dict(ccdnum=ccdnum))
+
+        # Generate gen3 files into local directory.
+        myLinearity = Linearizer(table=lsstTable)
+        for i, ampName in enumerate("AB"):
+            myLinearity.ampNames.append(ampName)
+            myLinearity.linearityType[ampName] = 'LookupTable'
+            myLinearity.linearityCoeffs[ampName] = np.array([i, 0])
+            myLinearity.linearityBBox[ampName] = detector.getAmplifiers()[i].getBBox()
+            myLinearity.fitParams[ampName] = np.array([])
+            myLinearity.fitParamsErr[ampName] = np.array([])
+            myLinearity.fitChiSq[ampName] = np.nan
+        myLinearity.updateMetadata(camera=camera, detector=detector,
+                                   CALIBDATE='1970-01-01T00:00:00',
+                                   setCalibId=True)
+        myLinearity.hasLinearity = True
+
+        outDir = os.path.join(getPackageDir('obs_decam'), 'decam', 'CALIB', 'linearity',
+                              detector.getName().lower())
+        if os.path.exists(outDir) and not force:
+            print("Output directory %r exists; use --force to replace" % (outDir, ))
```

Comment:
Would this happen if you already ingested curated calibs once and tried to do so again? At what point could a user actually set `--force`?

Reply:
I added `--force` to duplicate how the crosstalk script works.  Both are writing gen3 "curated" calibrations that can then be ingested into a repository (but not automatically doing that ingestion).  It was useful when I was debugging the script, and likely will only be useful in such cases.  If there's an update of the linearity data, the correct solution is to update the CALIBDATE to reflect when the new calibration should be used.
Ingesting calibrations multiple times fails in gen3 with a "ConflictingDefinitionError".  I don't think this raises in gen2, although it should replace the old with the new.